presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Insert.java

```diff
@@ -0,0 +1,102 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests.sqlserver;
+
+import com.teradata.tempto.AfterTestWithContext;
+import com.teradata.tempto.BeforeTestWithContext;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.Requirement;
+import com.teradata.tempto.RequirementsProvider;
+import com.teradata.tempto.configuration.Configuration;
+import com.teradata.tempto.query.QueryResult;
+import io.airlift.log.Logger;
+import org.testng.annotations.Test;
+
+import java.sql.Date;
+import java.sql.SQLException;
+
+import static com.facebook.presto.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static com.facebook.presto.tests.TestGroups.SQL_SERVER;
+import static com.facebook.presto.tests.sqlserver.SqlServerDataTypesTableDefinition.INSERT_TABLE_NAME;
+import static com.facebook.presto.tests.sqlserver.SqlServerDataTypesTableDefinition.SQLSERVER_INSERT;
+import static com.facebook.presto.tests.sqlserver.TestConstants.KEY_SPACE;
+import static com.facebook.presto.tests.utils.QueryExecutors.onPresto;
+import static com.facebook.presto.tests.utils.QueryExecutors.onSqlServer;
+import static com.teradata.tempto.assertions.QueryAssert.Row.row;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.fulfillment.table.TableRequirements.immutableTable;
+import static java.lang.String.format;
+
+public class Insert
+        extends ProductTest
+        implements RequirementsProvider
+{
+    @Override
+    public Requirement getRequirements(Configuration configuration)
+    {
+        return immutableTable(SQLSERVER_INSERT);
+    }
+
+    @BeforeTestWithContext
+    @AfterTestWithContext
+    public void dropTestTables()
+    {
+        try {
+            onPresto().executeQuery(format("DROP TABLE IF EXISTS %s", INSERT_TABLE_NAME));
+        }
+        catch (Exception e) {
+            Logger.get(getClass()).warn(e, "failed to drop table");
+        }
+    }
+
+    @Test(groups = {SQL_SERVER, PROFILE_SPECIFIC_TESTS})
+    public void testInsert()
+            throws SQLException
+    {
+        String sql = format(
+                "INSERT INTO %s.%s.%s values (BIGINT '%s', SMALLINT '%s', INTEGER '%s', DOUBLE '%s', " +
+                        "CHAR 'a   ', 'aa', DOUBLE '%s', DATE '%s')",
+                "sqlserver", KEY_SPACE, INSERT_TABLE_NAME, Long.valueOf("-9223372036854775807"), Short.MIN_VALUE, Integer.MIN_VALUE,
+                Double.MIN_VALUE, Double.MIN_VALUE, Date.valueOf("1970-01-01"));
+        onPresto().executeQuery(sql);
+
+        sql = format(
```

Comment:
Make it a separate test instead of combining them into one method.
Then it's more readable and there're no side effects from one test run affecting another.